### PR TITLE
APPSRE-3079: Improve image size/building

### PR DESCRIPTION
**Pull request by lechuk47**
## Overview
* Add a new qontract-reconcile-builder image with the required packages to build qonctract-reconcile application. This image will be used as a stage container to build the application 
* Improve qontract-reconcile-base image to just container the tooling required by qontract-reconcile app (terraform providers, binaries and packages)

## Notes
* Related PR: app-sre/qontract-reconcile#1879

### CODE REVIEW

File: qontract-reconcile-base/Dockerfile [hunk 1]
```diff
@@ -1,63 +1,91 @@
...
+    /usr/local/bin/amtool \
+    /usr/bin/skopeo \
+    /usr/local/bin/
 
-RUN mkdir -p ${TF_PLUGIN_CACHE_DIR}
-COPY --from=terraform-providers-downloader /tmp/terraform-providers/aws ${TF_PLUGIN_LOCAL_DIR}/plugins/registry.terraform.io/hashicorp/aws/
-COPY --from=terraform-providers-downloader /tmp/terraform-providers/time ${TF_PLUGIN_LOCAL_DIR}/plugins/registry.terraform.io/hashicorp/time/
-RUN find /.terraform.d/ -type d -exec chmod 0775 {} \;
+RUN mkdir -p ${TF_PLUGIN_CACHE_DIR} && \
+    microdnf upgrade -y && \
```
Comment (rporres) [inline qontract-reconcile-base/Dockerfile:83]: this is cool, so every time we build a new image we should get the latest `ubi-minimal:8.4` packages, right?

File: qontract-reconcile-builder/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        python39 \
+        gnupg2 \
+        python3{,9}-devel \
+        make \
+        gcc \
+        gcc-c++ \
+        openssl \
+        glibc-langpack-en && \
+    dnf clean all && \
+    update-alternatives --set python3 /usr/bin/python3.9
```
Comment (rporres) [inline qontract-reconcile-builder/Dockerfile:15]: nit: please add an ending new line

File: qontract-reconcile-base/Dockerfile [hunk 3]
```diff
@@ -1,63 +1,91 @@
-FROM registry.access.redhat.com/ubi8/ubi:8.4 AS terraform-providers-downloader
+FROM registry.access.redhat.com/ubi8/ubi:8.4 AS downloader
+
+RUN dnf -y install unzip glibc-langpack-en;
```
[the remark this answers is not in the document]
Comment (lechuk47) [reply qontract-reconcile-base/Dockerfile]: fixed

File: qontract-reconcile-builder/Dockerfile (hunk 2 above)
Comment (lechuk47) [reply qontract-reconcile-builder/Dockerfile]: fixed

**Pull request merged**